sway-fmt-v2/src/fmt.rs

```diff
@@ -46,19 +50,23 @@ impl Formatter {
 
         // Formatted code will be pushed here with raw newline stlye.
         // Which means newlines are not converted into system-specific versions by apply_newline_style
-        let mut raw_formatted_code = String::new();
+        // Use the length of src as a hint of the memory size needed for raw_formatted_code,
+        // which will reduce the number of reallocations
+        let mut raw_formatted_code = String::with_capacity(src_len);
```

Comment:
I don't think this will be necessary since our formatting will do more than just simple whitespace augmentation. We may end up reformatting things like long `generics` with a `where` clause which could cause collisions in the future if we give `raw_formatted_code` an initial capacity.

Reply:
Hmm, I don't understand what the "collision" means here, would you mind explain it more?

Reply:
My mistake, I was really tired yesterday and somehow thought `with_capacity` limited the capacity which isn't true at all 🤦🏻